gossipd/routing.c

```diff
@@ -442,6 +442,10 @@ static bool node_announce_predates_channels(const struct node *node)
 		if (!is_chan_public(c))
 			continue;
 
+		/* Zombies don't count! */
+		if (is_chan_zombie(c))
+			continue;
```

Comment:
By not moving the node announcement here, our logs get bad order messages.
`lightningd-2 2023-02-15T01:21:34.345Z DEBUG   gossipd: Bad gossip order: WIRE_NODE_ANNOUNCEMENT before announcement`
`routing_add_channel_update` can move on, and gossmap is blissfully ignorant, but if there's not a node to attach the node announcement to, that may be an issue.